### ISSUE TITLE
Comparing `char`s in constant expression produces incorrect warning

### ISSUE THREAD

**Issue by matthewjasper** [rust-lang/rust#40970]
The following code
```rust
const X: bool = 'A' == 'B';
```
produces a warning ([playground](https://is.gd/vSjGct))
```
warning: constant evaluation error: bad operands for binary. This will become a HARD ERROR in the future
 --> <anon>:4:21
  |
4 |     const X: bool = 'A' == 'B';
  |                     ^^^^^^^^^^
  |
  = note: #[warn(const_err)] on by default
```
but `X` appears to be correctly set to `false`.
